addons/project/views/project_views.xml

```diff
@@ -1394,7 +1438,9 @@
                           event_open_popup="true" quick_add="false" show_unusual_days="True"
                           js_class="project_calendar"
                           scales="month,year">
+                    <field name="allow_milestones" invisible="1" />
                     <field name="project_id" widget="project_private_task"/>
+                    <field name="milestone_id" attrs="{'invisible': [('allow_milestones', '=', False)]}" groups="project.group_project_milestone"/>
```

Comment:
```suggestion
                    <field name="milestone_id" attrs="{'invisible': [('allow_milestones', '=', False)]}"/>
```
I think the group is unnecessary here as it is already taken into account when computing `allow_milestones`